### PR TITLE
Improve docker build scripting

### PR DESCRIPTION
**Pull request by lukasz-walkiewicz**

### CODE REVIEW

File: presto/incremental.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+LABEL maintainer="https://www.starburstdata.com/"
+
+ARG presto_version=323-e.0
+ARG dist_location=""
+COPY ./installdir /installdir
+
+RUN set -xeu && \
+    dist="$(/installdir/find-dist-location.sh "${dist_location}" "${presto_version}")" && \
```
Comment (sopel39) [inline presto/incremental.Dockerfile]: remove this line as it's noop

File: presto/incremental.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Warning: This is for development purposes only.
+
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+LABEL maintainer="https://www.starburstdata.com/"
+
+ARG presto_version=323-e.0
```
Comment (sopel39) [inline presto/incremental.Dockerfile]: I would remove defaults here as it's supposed to use by script

**Comment by lukasz-walkiewicz** [comment]
> small comments. We probably should also have similar script for releasing

Sure, I'll do that in the next iteration.

File: .gitignore [hunk 3]
```diff
@@ -2,3 +2,4 @@
 *~
 presto/installdir/presto-cli-*-executable.jar
 presto/installdir/presto-server-rpm-*.x86_64.rpm
+.DS_Store
```
Comment (kokosing) [inline .gitignore]: what is this?

File: presto/Dockerfile [hunk 4]
```diff
@@ -4,13 +4,13 @@ LABEL maintainer="https://www.starburstdata.com/"
 ARG presto_version=323-e.0
 ARG dist_location=""
 
-COPY ./installdir /installdir
```
Comment (kokosing) [inline presto/Dockerfile]: mention motivation behind this change in the commit message

File: presto/build-image.sh [hunk 5]
```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Gets the command name without path
+cmd(){ echo `basename $0`; }
+
+function usage()
+{
+echo "\
```
Comment (kokosing) [inline presto/build-image.sh]: indentation?

File: presto/build-image.sh [hunk 6]
```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
```
Comment (kokosing) [inline presto/build-image.sh]: add `set -euo pipefail`

File: presto/build-image.sh [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+# Gets the command name without path
+cmd(){ echo `basename $0`; }
```
Comment (kokosing) [inline presto/build-image.sh]: prefix with `function`.

No need to wrap `basename` with `echo`.

Format this as
```
cmd()
{ 
    basename $0
}
```

**Comment by sopel39** [comment]
still lgtm

File: presto/build-image.sh [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+cp $PRESTO_RPM installdir
+cp $PRESTO_CLI installdir
+
+IMAGE_NAME=starburstdata/presto:${PRESTO_VERSION}
+
+if [ "${INCREMETAL}" = true ] && [[ $(docker image list -q ${IMAGE_NAME}) ]]; then
+  echo "Running incremetal build..."
+  docker build . \
+    --build-arg "presto_version=${PRESTO_VERSION}" \
+    --build-arg "BASE_IMAGE=${IMAGE_NAME}" \
```
Comment (kokosing) [inline presto/build-image.sh]: what if `IMAGE_NAME` is not yet build?

File: presto/incremental.Dockerfile [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+ARG presto_version
+ARG dist_location
+
+COPY ./installdir /installdir
+
+RUN set -xeu && \
+    yum -y install rsync && \
+    mkdir /tmp/presto && cd /tmp/presto && rpm2cpio "${dist_location}/presto-server-rpm-${presto_version}.x86_64.rpm" | cpio -idm && \
+    rsync -ir --checksum /tmp/presto/ / && \
```
Comment (kokosing) [inline presto/incremental.Dockerfile]: `/tmp/presto/` comes from `cpio`?

File: presto/incremental.Dockerfile [hunk 10]
```diff
@@ -0,0 +1,23 @@
+# Warning: This is for development purposes only.
```
Comment (kokosing) [inline presto/incremental.Dockerfile]: How much faster is this?

Comment (sopel39) [reply presto/incremental.Dockerfile]: It's 2GB vs few tens of MBs. It's much, much faster when uploading to ECR

File: .gitignore (hunk 3 above)
Comment (lukasz-walkiewicz) [reply .gitignore]: https://en.wikipedia.org/wiki/.DS_Store

File: presto/build-image.sh (hunk 6 above)
Comment (lukasz-walkiewicz) [reply presto/build-image.sh]: See line 53

File: presto/build-image.sh (hunk 8 above)
Comment (lukasz-walkiewicz) [reply presto/build-image.sh]: Then we won't meet the condition above this line won't execute.

File: presto/incremental.Dockerfile (hunk 9 above)
Comment (lukasz-walkiewicz) [reply presto/incremental.Dockerfile]: from `mkdir /tmp/presto && cd /tmp/presto && rpm2cpio "${dist_location}/presto-server-rpm-${presto_version}.x86_64.rpm`

File: presto/build-image.sh (hunk 6 above)
Comment (kokosing) [reply presto/build-image.sh]: I guess it should be at the top. It is common header of bash scripts.

File: presto/build-image.sh [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+-r, --rpm; Set presto server RPM file location
+-c, --cli; Set presto CLI executable jar location
+-v, --version; Set presto version
+-i, --incremental; Allow incremetal build
+" | column -t -s ";"
+}
+
+INCREMETAL=false
+
+options=`getopt -o r:c:v:i --long rpm:,cli:,version:,incremental -n 'parse-options' -- "$@"`
```
Comment (kokosing) [inline presto/build-image.sh]: use `$(..)` instead of backticks

File: presto/build-image.sh [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+if [ $? != 0 ] ; then echo "Failed parsing options." >&2 ; exit 1 ; fi
```
Comment (kokosing) [inline presto/build-image.sh]: maybe inline `$?`. Also move `echo`, `exit` and `fi` into separate lines

File: presto/build-image.sh [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+PRESTO_RPM_BASENAME=$(basename $PRESTO_RPM)
+PRESTO_CLI_BASENAME=$(basename $PRESTO_CLI)
+
+function cleanup {
+  rm -f "installdir/${PRESTO_RPM_BASENAME}"
+  rm -f "installdir/${PRESTO_CLI_BASENAME}"
+}
+trap cleanup EXIT
+
+cp $PRESTO_RPM installdir
```
Comment (kokosing) [inline presto/build-image.sh]: use `""`

**Pull request merged**